Add Lasagna to the game

alphanerdd opened the pull request:
This is my first PR, and also my first time working with BYOND, so go easy on me. Also first time really working with GitHub, so let me know if I did any of this wrong.

## About The Pull Request

This PR adds Lasagna as a craftable food item into the game, as well as a sprite and crafting recipe.

![image](https://user-images.githubusercontent.com/60521518/96322053-97953e80-0fe5-11eb-9ad4-7ad748c612bf.png)
![image](https://user-images.githubusercontent.com/60521518/96322056-99f79880-0fe5-11eb-96fa-99ec0b774d3a.png)
![image](https://user-images.githubusercontent.com/60521518/96322060-9c59f280-0fe5-11eb-88ce-a092801b9450.png)
![image](https://user-images.githubusercontent.com/60521518/96322066-9ebc4c80-0fe5-11eb-9aea-d7f03a6bfa47.png)

## Why It's Good For The Game

I was inspired to add Lasagna to the game after viewing a lasagna appreciation forum post, where i discovered there is no lasagna currently in the game. How can we properly appreciate Lasagna without being able to make it in our kitchens?

Overall a pretty inconsequential addition, but I think having lasagna in the game is an overall positive.

## Changelog
:cl:
add: Adds lasagna as a food item, including a craftable recipe and sprite. 
/:cl:

alphanerdd committed:
Lasagna

Adds Lasagna

alphanerdd committed:
Update recipes_meat.dm

alphanerdd commented:
After discussion over discord, i have changed the recipe to use spaghetti over buns

Jared-Fogle commented:
Changelogs are for players, not for coders. Your changelog should say just what is necessary to players.

Jared-Fogle commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_meat.dm:
```diff
@@ -227,3 +227,14 @@
+		/obj/item/food/meat/cutlet = 2,
+		/obj/item/reagent_containers/food/snacks/grown/tomato = 1,
+		/obj/item/reagent_containers/food/snacks/cheesewedge = 2,
+		/obj/item/food/spaghetti
```
This right here is why you shouldn't do "web edits", AKA using github.com to make commits.

It's a way of showing to us that you didn't test your code. In this case, you web edited it to use spaghetti instead of whatever it was before, but forgot to put the `= 1` necessary to make it actually work.

```suggestion
		/obj/item/food/spaghetti = 1
```

alphanerdd replied:
Thanks for pointing that out. I figured it would be fine, but obviously not lol. Will make sure to double check in the future. Also will change the changelog to be for players.

alphanerdd committed:
Update code/modules/food_and_drinks/recipes/tablecraft/recipes_meat.dm

Co-authored-by: Jared-Fogle <35135081+Jared-Fogle@users.noreply.github.com>

necromanceranne commented:
dare I say, using a roller pin on raw spaghetti should make raw lasagna sheets

Farquaar commented:
Speedmerge

Wallemations commented:
I agree with lasagna sheets needing to be added, I wish to create multiple types of lasagna as well as lasagna floors and tables.
Although having just lasagna is fine too.

Farquaar commented:
> 
> 
> I agree with lasagna sheets needing to be added, I wish to create multiple types of lasagna as well as lasagna floors and tables.

I hate mondays

Ghommie commented:
No butter, flour and milk for the bechamel sauce? I'm sad.

Qustinnus commented:
Hey,

awesome work on the PR. We're currently refactoring food to a new typepath. If you feel like you're up for it could you instead of making this a /obj/item/reagent_containers/food/snacks, make it a /obj/item/food ?

You're in the right file for making the food, but as you can see from the other objects in the file you're using theyre using a new typepath (/obj/item/food). Some of the variables in this type are named a bit different, and the food can't have value, but besides that it should be the same.

If you can't figure it out feel free to ask on the Discord, or on here.

Qustinnus commented:
and if you for some reason want to indulge the crazy people who want lasagna sheets I could explain how it could be done, but if you actually want to entertain those radicals is up to you.

Fikou commented:
please look what the actual changelog tags are, there isnt such a thing as additem

necromanceranne commented:
Oh when I said 'sheets' I didn't meant mat datum stuff I literally just meant turning one ingredient into another like you do with bread dough into buns. But I mean, lasagne mats sounds funny.

alphanerdd committed:
Updated food formating

Changes the lasagna from /obj/item/reagent_containers/food/snacks to  /obj/item/food.

alphanerdd commented:
> Hey,
> 
> awesome work on the PR. We're currently refactoring food to a new typepath. If you feel like you're up for it could you instead of making this a /obj/item/reagent_containers/food/snacks, make it a /obj/item/food ?
> 
> You're in the right file for making the food, but as you can see from the other objects in the file you're using theyre using a new typepath (/obj/item/food). Some of the variables in this type are named a bit different, and the food can't have value, but besides that it should be the same.
> 
> If you can't figure it out feel free to ask on the Discord, or on here.

Thanks for the feedback! I think i was able to figure out the formatting. When i compiled and ran it, everything seemed good, but if there's something i missed,  please let me know. As for the lasagna tiles, that seems a bit out of my league for the time being, although it might be something fun to try in the future.

Jared-Fogle commented:
Going to wait on @Twaticus to +1 the art.

Tharcoonvagh commented:
![kqg8tk1bgqy11.jpg](https://user-images.githubusercontent.com/54288658/96362258-59566700-112c-11eb-8b50-30aa9bff98b1.jpg)

Qustinnus commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_meat.dm:
```diff
@@ -227,3 +227,14 @@
+		/obj/item/food/spaghetti = 1
+	)
+	result = /obj/item/food/lasagna
+	subcategory = CAT_MEAT
```
pun intended

TacticalSpacePilot commented:
It doesn't look like lasagna in my opinion here's what it could look like 

https://wiki.yogstation.net/wiki/File:Lasagna.png

Twaticus reviewed (approved):
>cutlet, tomato, cheese wedge, bun.

what a sad sounding lasagna

alphanerdd commented:
> > cutlet, tomato, cheese wedge, bun.
> 
> what a sad sounding lasagna

The screenshot is old, the recipe got changed to use spaghetti instead of pasta. Notably less sad, though still a struggle.

alphanerdd commented:
> It doesn't look like lasagna in my opinion it looks like dollar store microwave lasagna, here's what it could look like
> 
> https://wiki.yogstation.net/wiki/File:Lasagna.png

Yeah the sprite isn't amazing. Definitely hits that probably edible vibe you get from like gas station food. Tbh its my first time spriting as well, and I'm by no means an artist. It's honestly the best I could do.

The real question is how did Yogstation get lasagna before TG?

Jared-Fogle commented:
Thanks!

BootlegBow commented:
beautiful
